app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -121,7 +167,38 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
-      logger.d("StateFragment", "getCurrentState: ${result.state.name}")
+      if (result.hasState()) {
+        itemList.clear()
+        currentEphemeralState.set(result)
+
+        updateDummyStateName()
+
+        val interactionId = result.state.interaction.id
+        val hasPreviousState = result.hasPreviousState
+        var hasNextState = false
+        hasGeneralContinueButton = false
+
+        if (!result.terminalState) {
+          if (result.stateTypeCase.number == EphemeralState.COMPLETED_STATE_FIELD_NUMBER
+            && !oldStateNameList.contains(currentEphemeralState.get()!!.state.name)
```

Comment:
Here & below: let's just use ``result`` directly instead of refetching the value from the ObservableField (then you won't need to de-null it).

Reply:
I was not sure whether you wanted me to remove `currentEphemeralState` entirely or you just wanted me to use `result` in this function only.

So for now, I am removing `currentEphemeralState` from `ObservableField` because we dont need this to be `ObservableField` and we can directly use this as 

`private lateinit var currentEphemeralState: EphemeralState`